src/main/java/org/cryptomator/common/settings/Settings.java

```diff
@@ -134,6 +131,15 @@ public static Settings create(Environment env) {
 		lastUpdateCheck.addListener(this::somethingChanged);
 	}
 
+	@SuppressWarnings("deprecation")
+	private void migratePortToVaultSettings(SettingsJson json) {
+		if(json.port != 0){
+			for (VaultSettings vaultSettings : directories) {
+				vaultSettings.port.set(json.port);
+			}
+		}
+	}
```

Comment:
What is the strategy here? Do we want multiple isolated servers? Then they must not use the same port. Or open multiple ports on the same server?

This adds a lot of complexity and potential for future support requests. I am not sure it is a good idea to have conflicting settings for different vaults

Reply:
And this is not even taking into account that different mount providers requiring a TCP port are mixed.